### DISCUSSION TITLE
termux-ui-bug

### DISCUSSION THREAD

**Discussion by learningBASh** [termux/termux-app#3831] [Q&A]
termux-ui-bug - edit of line and word count script exersize of lesson 3 (from a bash guide) ran /in bash/ and did not quit while showing nothing - needing to be killed- the script was then begun to be edited to match the practice solution and after a ctrl+k cut the entire file seemed to reformat replacing newlines with spaces of some sort after such queer happening I continued to edit the file to match the solution and when done so the script was rerun with the same happen of no output needing ctrl+c kill. 
 steps taken: I deleted the file and rewrote it now working. the happening I guess was some invisible characters messing up the running although the nano syntax parser seemed to read the code well enough.
(unsolved) this bug should be investigated further with nano and termux support groups to prevent it from happen again and familiarize with every bit of code writing with said tools.
Q. How can I find out what causes a reformatting of a text file possibly having to do witg termux-ui and ctrl+k?
A.

**Comment by TomJo2000** [comment]
Could you please provide a link to the guide you were following, for reference?

It's likely your text was being reformatted due to a setting in your `nano` configuration.
If I had to make a guess what caused it, it probably expanded tabs to spaces on save and thus changed the appearance.

**Reply by learningBASh** [reply]
I was looking into reiterating the bash tutorial guide using the github.dev editor. I can link my bug report about github.dev web editor which is posted as a General discussion to GitHub/community because I did not find a code repo by searching for "github.dev". If the tools were working I can add the entire tutorial class as a repo and link you to it. Cheers :-) @TomJo2000

**Reply by TomJo2000** [reply]
github.dev is a online version of the VS Code editor, it's not a tutorial.

**Reply by learningBASh** [reply]
Here is my bug report on github.dev https://github.com/orgs/community/discussions/107473#discussion-6204395

**Comment by learningBASh** [comment]
> Could you please provide a link to the guide you were following, for reference?
> 

the guide is irrelevent

> It's likely your text was being reformatted due to a setting in your `nano` configuration. If I had to make a guess what caused it, it probably expanded tabs to spaces on save and thus changed the appearance.

Your guess is a bit off as you see I specified newlines and not tabs, although posting this to termux nano package repo may prove useful can you link the proper repo for termux package nano?

**Reply by TomJo2000** [reply]
The build files and issue tracker for all Termux packages are housed at:
https://github.com/termux/termux-packages


To quote from the README of this repo, which is for the core App itself ***not*** the packages, such as `nano` or `command-not-found`:
> [!IMPORTANT]
> Note that this repository is for the app itself (the user interface and the terminal emulation). For the packages installable inside the app, see termux/termux-packages.

**Comment by learningBASh** [comment]
> To quote from the README of this repo, which is for the core App itself _**not**_ the packages, such as `nano` or `command-not-found`:
> 
> Important
> 
> Note that this repository is for the app itself (the user interface and the terminal emulation). For the packages installable inside the app, see termux/termux-packages.

the special CTRL button is specificly a part of the core app. The base packages **could be both**    ( those packages are not downloaded )  The packages included in the app are certainly part of the app. Those packages do fall under termux-app the app itself.
Q. How can I find out what causes a reformatting of a text file possibly having to do with termux-ui and ctrl+k?
A. possedebit this is the terminal emulation interfering between the user and apps.
An answer may begin like so:
A. To find out if the termux ui special keys input is sending any inadvertent signals run termux in debug mode by doing such and such then looking at the special keys log here. When such has been ruled out consider next discussing this with nano support. ( then uou could maybe change your nametag to nano support ) and the signsls to look for would be such and such that nano will reformat texts when recieved.
OR
A. There is no way to find out if termux ui is bugged.

Your answer seems to be dodgey, again, as I said **newlines** and you reply about tabs, **dodgey**. You also seem to be cross-posting without referencing another discussion thread you and I had.
 Dodgey support pointing me away. Before I would cross-post to termux-packages I need to rule out possedebit of the cause being at the base. Do notice that I labled this a BUG where an emulation which interferes between the user and nano is a "bug". bugged?
